presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -320,6 +369,64 @@ public static Domain getDomain(Type type, DictionaryDescriptor dictionaryDescrip
         return Domain.all(type);
     }
 
+    @VisibleForTesting
+    public Domain getDomain(Type type, long rowCount, ColumnIndex columnIndex, ParquetDataSourceId id, RichColumnDescriptor descriptor, boolean failOnCorruptedParquetStatistics)
+            throws ParquetCorruptionException
+    {
+        if (columnIndex == null) {
+            return Domain.all(type);
+        }
+
+        String columnName = descriptor.getPrimitiveType().getName();
+
+        if (isCorruptedColumnIndex(columnIndex)) {
+            if (failOnCorruptedParquetStatistics) {
+                failWithCorruptionException(failOnCorruptedParquetStatistics, columnName, id, columnIndex);
+            }
+            else {
+                return Domain.all(type);
+            }
+        }
+
+        if (isEmptyColumnIndex(columnIndex)) {
+            return Domain.all(type);
+        }
+
+        long totalNullCount = columnIndex.getNullCounts().stream().reduce(0L, (a, b) -> a + b);
+        if (totalNullCount == rowCount) {
+            return Domain.onlyNull(type);
+        }
+
+        boolean hasNullValue = totalNullCount > 0;
+
+        if (descriptor.getType().equals(PrimitiveTypeName.BOOLEAN)) {
+            // After row-group filtering for boolean, page filtering shouldn't do more
+            return Domain.all(type);
+        }
+
+        if (descriptor.getType().equals(PrimitiveTypeName.INT32) || descriptor.getType().equals(PrimitiveTypeName.INT64) || descriptor.getType().equals(PrimitiveTypeName.FLOAT)) {
```

Comment:
is FLOAT min max converting to Longs?
static import PrimitiveTypeName.INT32, PrimitiveTypeName.INT64, etc.

Reply:
From my earlier test, yeah we convert it to long. 

Fixed the static imports.